pkg/virt-controller/services/template.go

```diff
@@ -1118,6 +1118,9 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 
 	pod.Spec.SchedulerName = vmi.Spec.SchedulerName
 
+	enableServiceLinks := false
+	pod.Spec.EnableServiceLinks = &enableServiceLinks
```

Comment:
While we don't use these env vars by default for anything that I'm aware of, I'm unsure if anyone has created any KubeVirt extensions (like the hooks api) which might somehow depend on this.

Maybe we should consider a cluster wide setting, something like `enableWorkloadServiceLinks` on the KubeVirt CRD. That would likely go in this struct, https://github.com/kubevirt/kubevirt/blob/27c2e7d02afbe17af95f031264dade8c24fa59b9/staging/src/kubevirt.io/client-go/api/v1/types.go#L1443

Reply:
It is unlikely, but still sounds good. Let's disable them by default but allow people to enable it still.

Reply:
I'm regretfully not adept enough in Kubevirt / Go development to know how to orchestrate this and generate the CRD etc. I would be very surprised if someone uses a service link in a side car, but that's just me.

Reply:
I think i misunderstood what happenes when this feature is disabled. As long as it doesn't remove the standard KUBERNETES_SERVICE env vars, like

```KUBERNETES_SERVICE_HOST=10.0.0.1
KUBERNETES_SERVICE_PORT=443
KUBERNETES_SERVICE_PORT_HTTPS=443```

then i'm fine with disabling it entirely

Reply:
Those are still there
```
KUBERNETES_SERVICE_PORT_HTTPS=443
KUBERNETES_SERVICE_PORT=443
KUBERNETES_PORT_443_TCP=tcp://10.135.192.1:443
KUBERNETES_PORT_443_TCP_PROTO=tcp
KUBERNETES_PORT_443_TCP_ADDR=10.135.192.1
KUBERNETES_SERVICE_HOST=10.135.192.1
KUBERNETES_PORT=tcp://10.135.192.1:443
KUBERNETES_PORT_443_TCP_PORT=443
```

Reply:
Your reasoning makes perfect sense. Could you sign your commit (`-s`), so that we can merge it? Thanks for the PR.

Reply:
On it

Reply:
Done and force pushed.